multi/stm32l4-multi/libmulti/libuart.c

```diff
@@ -481,18 +481,22 @@ static void libuart_infiniteRxHandler(void *arg, int type)
 	size_t cpysz;
 	char *rxbuf = (char *)ctx->data.dma.rxbuf;
```

Comment:
didn't read the code to check if rxbuf is needed to really be volatile, buuut:

 - is volatile dropped here consciously? If so - maybe add explicit comment?
 - please note that `memcpy` is not guaranteed to work with volatile memory but probably ok in this use case (assuming compiler won't optimize in some strange way)?

Reply:
Welp. I have many more questions here. 

- If DMA write data to fifo, and interrupt gathers data from that fifo, is it required to make this fifo volatile ?
- As memcpy not working for volatile types, should it be implemented as loop?
- As we should have fast and optimized cpy, should we do it as some assembly? 

In this change I added volatile for ctx->data.dma.rxbuf pointer (not data!)  which improves situation. As i write in PR it not resolving issue fully, it's rather obvious things that improves current situation (synchronisation is no longer done on non-volatile pointer) . So I would merge it as-is and do more work in the future, after more analysis.